posthog/temporal/batch_exports/bigquery_batch_export.py

```diff
@@ -276,7 +276,17 @@ async def amerge_person_tables(
 
         merge_query = f"""
         MERGE `{final_table.full_table_id.replace(":", ".", 1)}` final
-        USING `{stage_table.full_table_id.replace(":", ".", 1)}` stage
+        USING (
+            SELECT * FROM
+            (
+              SELECT
+              *,
+              ROW_NUMBER() OVER (PARTITION BY {",".join(field.name for field in merge_key)}) row_num
+            FROM
+              `{stage_table.full_table_id.replace(":", ".", 1)}`
+            )
+            WHERE row_num = 1
+        ) stage
```

Comment:
`SELECT DISTINCT` does not work with JSON columns, and there is no "DISTINCT BY"